### PR TITLE
MODPUBSUB-217 Replace Future<Boolean> by Future<Void> in KafkaTopicService

### PR DESCRIPTION
**Pull request by sherzodn**
https://issues.folio.org/browse/MODPUBSUB-217
Replace `Future<Boolean>` by `Future<Void>` for
- KafkaTopicService
- MessagingModuleService
-  PublisherService

### CODE REVIEW

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 1]
```diff
@@ -58,9 +59,9 @@
   private static final int RETRY_NUMBER = Integer.parseInt(MODULE_SPECIFIC_ARGS.getOrDefault("pubsub.delivery.retry.number", "5"));
 
   public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
-                                  @Autowired KafkaConfig kafkaConfig,
-                                  @Autowired SecurityManager securityManager,
-                                  @Autowired Cache cache) {
+    @Autowired KafkaConfig kafkaConfig,
+    @Autowired SecurityManager securityManager,
+    @Autowired Cache cache) {
```
Comment (julianladisch) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java:64]: Can you add more indentation?

**Review by alexanderkurash — APPROVED** [review]

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/SubscribersApiTest.java [hunk 2]
```diff
@@ -288,4 +253,14 @@ private EventDescriptor postEventDescriptor(EventDescriptor eventDescriptor) {
...
   }
+
+  private void postDeclareSubscriber(SubscriberDescriptor subscriberDescriptor) {
+    RestAssured.given()
+      .spec(spec)
+      .body(JsonObject.mapFrom(subscriberDescriptor).encode())
+      .when()
+      .post(EVENT_TYPES_PATH + DECLARE_SUBSCRIBER_PATH)
+      .then()
+      .statusCode(anyOf(is(HttpStatus.SC_CREATED), anyOf(is(HttpStatus.SC_INTERNAL_SERVER_ERROR))));
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/test/java/org/folio/rest/impl/SubscribersApiTest.java:264]: Can you explain this change? Why did it changed to this from `.statusCode(HttpStatus.SC_CREATED);` ?

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java [hunk 3]
```diff
@@ -284,13 +244,7 @@ public void shouldDeletePublisherOnDelete() {
       .withEventDescriptors(Collections.singletonList(createdEventDescriptor))
       .withModuleId("test-module-1.0.0");
 
-    RestAssured.given()
-      .spec(spec)
-      .body(JsonObject.mapFrom(publisherDescriptor).encode())
-      .when()
-      .post(EVENT_TYPES_PATH + DECLARE_PUBLISHER_PATH)
-      .then().log().all()
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java:292]: Did we loose `.log().all()` when extracting method?

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java [hunk 4]
```diff
@@ -332,4 +286,14 @@ private EventDescriptor postEventDescriptor(EventDescriptor eventDescriptor) {
...
   }
+
+  private void postDeclarePublisher(PublisherDescriptor publisherDescriptor) {
+    RestAssured.given()
+      .spec(spec)
+      .body(JsonObject.mapFrom(publisherDescriptor).encode())
+      .when()
+      .post(EVENT_TYPES_PATH + DECLARE_PUBLISHER_PATH)
+      .then()
+      .statusCode(anyOf(is(HttpStatus.SC_CREATED), anyOf(is(HttpStatus.SC_INTERNAL_SERVER_ERROR))));
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java:297]: Why did it changed to this from `.statusCode(HttpStatus.SC_CREATED);` ?

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/SubscribersApiTest.java (hunk 2 above)
Comment (sherzodn) [reply mod-pubsub-server/src/test/java/org/folio/rest/impl/SubscribersApiTest.java:264]: the same reason as commented above

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java (hunk 3 above)
Comment (sherzodn) [reply mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java:292]: should I revert back to the previous state?

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java (hunk 1 above)
Comment (sherzodn) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: done

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java [hunk 5]
```diff
@@ -15,7 +15,7 @@
    *
    * @param eventTypes list of event types that specified module is subscribing to receive
    * @param params     Okapi connection params
-   * @return future with true if succeeded
+   * @return a future completed with a result
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java]: fixed

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 6]
```diff
@@ -58,9 +59,9 @@
   private static final int RETRY_NUMBER = Integer.parseInt(MODULE_SPECIFIC_ARGS.getOrDefault("pubsub.delivery.retry.number", "5"));
 
   public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: done

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java (hunk 3 above)
Comment (sherzodn) [reply mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java]: fixed

File: mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java (hunk 4 above)
Comment (sherzodn) [reply mod-pubsub-server/src/test/java/org/folio/rest/impl/PublishersApiTest.java]: error is recovered as we agreed

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaTopicServiceImpl.java [hunk 7]
```diff
@@ -30,20 +30,13 @@ public KafkaTopicServiceImpl(@Autowired KafkaAdminClient kafkaAdminClient, @Auto
...
-        LOGGER.info("Created topics: [{}]", StringUtils.join(eventTypes, ","));
-        promise.complete(true);
-      } else {
-        LOGGER.info("Some of the topics [{}] were not created. Cause: {}", StringUtils.join(eventTypes, ","), ar.cause().getMessage());
-        promise.complete(false);
-      }
-    });
-    return promise.future();
+    return kafkaAdminClient.createTopics(topics)
+      .onSuccess(x -> LOGGER.info("Created topics: [{}]", StringUtils.join(eventTypes, ",")))
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaTopicServiceImpl.java]: done

File: mod-pubsub-server/src/main/java/org/folio/services/publish/PublishingService.java [hunk 8]
```diff
@@ -13,8 +13,8 @@
    *
    * @param event    {@link Event} to send
    * @param tenantId tenant id
-   * @return future with true if succeeded
+   * @return a future completed with a result
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-pubsub-server/src/main/java/org/folio/services/publish/PublishingService.java]: fixed

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**